sdk/go/common/util/gitutil/git.go

```diff
@@ -264,176 +262,91 @@ type urlAuthParser struct {
 
 	// sshKeys memoizes keys we've loaded for given host URLs, to avoid needing to
 	// re-fetch public keys.
-	sshKeys map[string]transport.AuthMethod
+	sshKeys map[string]AuthMethod
 	// sshConfig allows us to inject config for testing.
-	sshConfig sshUserSettings
+	sshConfig      sshUserSettings
+	sshAgentBroker sshAgentBroker
 }
 
 // defaultURLAuthParser uses the host's SSH configuration.
 var defaultURLAuthParser = &urlAuthParser{
-	sshConfig: ssh_config.DefaultUserSettings,
+	sshConfig:      ssh_config.DefaultUserSettings,
+	sshAgentBroker: &defaultSSHAgentBroker{},
 }
 
+type TransportAuth struct {
+	transport.AuthMethod
+}
+
+// publicKeys implements AuthMethod.
+func (w *TransportAuth) publicKeys() ([]ssh.Signer, error) {
+	return nil, errors.New("publicKeys not implemented for wrappedTransportAuth")
+}
+
+var _ AuthMethod = (*TransportAuth)(nil)
```

Comment:
This wrapping seems unnecessary. I think you basically want something like [this](https://github.com/charmbracelet/wishlist/blob/cc7a23ba71506a7f7a7069077836895f6916affa/client_auth.go#L59-L103).

Reply:
Nice find! That looks great.

Reply:
Reading this PR carefully, it looks like it tries _either_ the SSH agent _or_ identity files, whereas this will try both. Does that make sense?

Reply:
```sh
$ 2>&1 ssh -T git@github.com -v | grep -E '(get_agent|attempt key)'
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: ssh_fetch_identitylist: agent contains no identities
debug1: Will attempt key: /home/friel/.ssh/id_rsa RSA SHA256:UBBHXEKfKyZrdxl452Q/3nHhheARRX71vs2BJCNgyyA
debug1: Will attempt key: /home/friel/.ssh/id_ecdsa ECDSA SHA256:NBoR0bSB8XjmvGcY2FB+R+Kj13dAU8JbhvLJEkSl/Rg
debug1: Will attempt key: /home/friel/.ssh/id_ecdsa_sk
debug1: Will attempt key: /home/friel/.ssh/id_ed25519 ED25519 SHA256:85BNsgWxy6RzcAFQvX6AEZl82C0AWzEnLlFHJQdGy6w
debug1: Will attempt key: /home/friel/.ssh/id_ed25519_sk
debug1: Will attempt key: /home/friel/.ssh/id_xmss
debug1: Will attempt key: /home/friel/.ssh/id_dsa
```

If I run `ssh-add`, I get:

```sh
$ 2>&1 ssh -T git@github.com -v | grep -E '(get_agent|attempt key)'
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: agent returned 3 keys
debug1: Will attempt key: /home/friel/.ssh/id_rsa RSA SHA256:UBBHXEKfKyZrdxl452Q/3nHhheARRX71vs2BJCNgyyA agent
debug1: Will attempt key: /home/friel/.ssh/id_ecdsa ECDSA SHA256:NBoR0bSB8XjmvGcY2FB+R+Kj13dAU8JbhvLJEkSl/Rg agent
debug1: Will attempt key: /home/friel/.ssh/id_ed25519 ED25519 SHA256:85BNsgWxy6RzcAFQvX6AEZl82C0AWzEnLlFHJQdGy6w agent
debug1: Will attempt key: /home/friel/.ssh/id_ecdsa_sk
debug1: Will attempt key: /home/friel/.ssh/id_ed25519_sk
debug1: Will attempt key: /home/friel/.ssh/id_xmss
debug1: Will attempt key: /home/friel/.ssh/id_dsa
```

So I do think we want to "combine" both local identity and ssh agent, as that appears to be what happens here. (Though it looks like `ssh` combines them, I don't think we need to try to inspect the signers to deduplicate as it did.)